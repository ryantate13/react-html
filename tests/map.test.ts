import {map} from '../src';

describe('map', () => {
    it('renders with children', () => {
        expect(map({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(map({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(map({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(map()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(map({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(map({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(map({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(map({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(map({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(map({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(map({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(map({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(map({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(map({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(map({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(map({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(map({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(map({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(map({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(map({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(map({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(map({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(map({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(map({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(map({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(map({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(map({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(map({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(map({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(map({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(map({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(map({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(map({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(map({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(map({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(map({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(map({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(map({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(map({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(map({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(map({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(map({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(map({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(map({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(map({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(map({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(map({translate: "no"})).toMatchSnapshot();
    });

    it('renders with name="TEST"', () => {
        expect(map({name: "TEST"})).toMatchSnapshot();
    });
});