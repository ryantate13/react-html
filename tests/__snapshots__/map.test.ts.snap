// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map renders with accessKey="a,b,c,d" 1`] = `
<map
  accessKey="a,b,c,d"
/>
`;

exports[`map renders with autoCapitalize="characters" 1`] = `
<map
  autoCapitalize="characters"
/>
`;

exports[`map renders with autoCapitalize="none" 1`] = `
<map
  autoCapitalize="none"
/>
`;

exports[`map renders with autoCapitalize="off" 1`] = `
<map
  autoCapitalize="off"
/>
`;

exports[`map renders with autoCapitalize="on" 1`] = `
<map
  autoCapitalize="on"
/>
`;

exports[`map renders with autoCapitalize="sentences" 1`] = `
<map
  autoCapitalize="sentences"
/>
`;

exports[`map renders with autoCapitalize="words" 1`] = `
<map
  autoCapitalize="words"
/>
`;

exports[`map renders with children 1`] = `
<map>
  TEST
</map>
`;

exports[`map renders with className="foo bar baz" 1`] = `
<map
  className="foo bar baz"
/>
`;

exports[`map renders with contentEditable="false" 1`] = `
<map
  contentEditable="false"
/>
`;

exports[`map renders with contentEditable="true" 1`] = `
<map
  contentEditable="true"
/>
`;

exports[`map renders with dir="auto" 1`] = `
<map
  dir="auto"
/>
`;

exports[`map renders with dir="ltr" 1`] = `
<map
  dir="ltr"
/>
`;

exports[`map renders with dir="rtl" 1`] = `
<map
  dir="rtl"
/>
`;

exports[`map renders with draggable="false" 1`] = `
<map
  draggable="false"
/>
`;

exports[`map renders with draggable="true" 1`] = `
<map
  draggable="true"
/>
`;

exports[`map renders with empty attributes 1`] = `<map />`;

exports[`map renders with hidden="false" 1`] = `
<map
  hidden={false}
/>
`;

exports[`map renders with hidden="true" 1`] = `
<map
  hidden={true}
/>
`;

exports[`map renders with id="TEST" 1`] = `
<map
  id="TEST"
/>
`;

exports[`map renders with inputMode="decimal" 1`] = `
<map
  inputMode="decimal"
/>
`;

exports[`map renders with inputMode="email" 1`] = `
<map
  inputMode="email"
/>
`;

exports[`map renders with inputMode="none" 1`] = `
<map
  inputMode="none"
/>
`;

exports[`map renders with inputMode="numeric" 1`] = `
<map
  inputMode="numeric"
/>
`;

exports[`map renders with inputMode="search" 1`] = `
<map
  inputMode="search"
/>
`;

exports[`map renders with inputMode="tel" 1`] = `
<map
  inputMode="tel"
/>
`;

exports[`map renders with inputMode="text" 1`] = `
<map
  inputMode="text"
/>
`;

exports[`map renders with inputMode="url" 1`] = `
<map
  inputMode="url"
/>
`;

exports[`map renders with is="foo" 1`] = `
<map
  is="foo"
/>
`;

exports[`map renders with itemID="https://google.com/" 1`] = `
<map
  itemID="https://google.com/"
/>
`;

exports[`map renders with itemProp="https://google.com/" 1`] = `
<map
  itemProp="https://google.com/"
/>
`;

exports[`map renders with itemRef="foo bar" 1`] = `
<map
  itemRef="foo bar"
/>
`;

exports[`map renders with itemScope="false" 1`] = `
<map
  itemScope={false}
/>
`;

exports[`map renders with itemScope="true" 1`] = `
<map
  itemScope={true}
/>
`;

exports[`map renders with itemType="https://google.com/" 1`] = `
<map
  itemType="https://google.com/"
/>
`;

exports[`map renders with lang="" 1`] = `
<map
  lang=""
/>
`;

exports[`map renders with lang="es-419" 1`] = `
<map
  lang="es-419"
/>
`;

exports[`map renders with name="TEST" 1`] = `
<map
  name="TEST"
/>
`;

exports[`map renders with slot="TEST" 1`] = `
<map
  slot="TEST"
/>
`;

exports[`map renders with spellCheck="false" 1`] = `
<map
  spellCheck="false"
/>
`;

exports[`map renders with spellCheck="true" 1`] = `
<map
  spellCheck="true"
/>
`;

exports[`map renders with style="[object Object]" 1`] = `
<map
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`map renders with tabIndex="1" 1`] = `
<map
  tabIndex={1}
/>
`;

exports[`map renders with title="TEST" 1`] = `
<map
  title="TEST"
/>
`;

exports[`map renders with translate="no" 1`] = `
<map
  translate="no"
/>
`;

exports[`map renders with translate="yes" 1`] = `
<map
  translate="yes"
/>
`;

exports[`map renders without attributes 1`] = `<map />`;
